control-announcements/src/main/java/org/oskari/announcements/actions/AnnouncementsHandler.java

```diff
@@ -0,0 +1,108 @@
+package org.oskari.announcements.actions;
+
+import fi.nls.oskari.control.RestActionHandler;
+import org.oskari.announcements.helpers.AnnouncementsParser;
+import org.oskari.announcements.helpers.Announcement;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.oskari.log.AuditLog;
+import fi.nls.oskari.util.JSONHelper;
+
+@OskariActionRoute("Announcements")
+public class AnnouncementsHandler extends RestActionHandler {
+    private static final Logger LOG = LogFactory.getLogger(AnnouncementsHandler.class);
+    private AnnouncementsService service = new AnnouncementsServiceMybatisImpl();
+
+    // Handle get announcements
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
+
+        try {
+            JSONObject result = new JSONObject();
+            if (params.getUser().isAdmin()) {
+                result = service.getAdminAnnouncements();
```

Comment:
What is the difference with admin announcements and regular ones? Would you want the admin to see the regular ones but be able to load the admin ones for an admin UI? If so you could use the same getter but add a parameter to toggle which response you would like (and also check that only admin can use that toggle).

Reply:
Answering myself, the difference is that admin gets all announcements and normal users just the current ones. So when logged in as admin the announcements don't work the same way as for other users. In my opinion this needs a parameter that is only allowed for admin users for getting all announcements.

Reply:
Is there a possible way to achieve that without turning this into PUT or POST ?

Reply:
Yes, you can send URL-parameters in GET requests as well: https://en.wikipedia.org/wiki/Query_string